software/controller/lib/hal/i2c.cpp

```diff
@@ -339,19 +338,13 @@ void STM32Channel::SetupDMAChannels(const DMA::Base dma) {
 
   for (auto &map : DmaMap) {
     if (dma == map.dma_base && i2c_ == map.i2c_base) {
-      auto *dma_register = DMA::get_register(dma);
-      dma_ = dma;
-      tx_channel_ = &dma_register->channel[static_cast<uint8_t>(map.tx_channel_id)];
-      rx_channel_ = &dma_register->channel[static_cast<uint8_t>(map.rx_channel_id)];
-
-      // Tell the STM32 that those two DMA channels are used for I2C
-      DMA::SelectChannel(dma, map.rx_channel_id, map.request_number);
-      DMA::SelectChannel(dma, map.tx_channel_id, map.request_number);
-
-      // configure both DMA channels to handle I²C transfers
-      ConfigureDMAChannel(rx_channel_, ExchangeDirection::Read);
-      ConfigureDMAChannel(tx_channel_, ExchangeDirection::Write);
+      tx_dma_.emplace(dma, map.tx_channel_id);
+      rx_dma_.emplace(dma, map.rx_channel_id);
 
+      tx_dma_->Initialize(map.request_number, &(i2c_->tx_data), DMA::ChannelDir::MemoryToPeripheral,
+                          1);
```

Comment:
I would maybe add some comment on what `1` is meant to be, i.e. what is the name of the parameter being set. Some for below. Maybe we could use a pre-defined constant?

Reply:
Will add it in the I2C PR, which also changes this line, for ease of rebasing.